e3sm_diags/driver/aerosol_aeronet_driver.py

```diff
@@ -0,0 +1,119 @@
+    """Runs the tropical cyclone analysis diagnostic.
+
+    :param parameter: Parameters for the run
+    :type parameter: CoreParameter
+    :raises ValueError: Invalid run type
+    :return: Parameters for the run
+    :rtype: CoreParameter
+    """
+    variables = parameter.variables
+    run_type = parameter.run_type
+    seasons = parameter.seasons
+
+    for season in seasons:
+        test_data = utils.dataset.Dataset(parameter, test=True)
+        parameter.test_name_yrs = utils.general.get_name_and_yrs(
+            parameter, test_data, season
+        )
+        parameter.ref_name_yrs = "AERONET (2006-2015)"
+
+        for var in variables:
+            logger.info("Variable: {}".format(var))
+            parameter.var_id = var
+
+            test = test_data.get_climo_variable(var, season)
+            test_site = interpolate_model_output_to_obs_sites(test, var)
+
+        if run_type == "model_vs_model":
+            ref_data = utils.dataset.Dataset(parameter, ref=True)
+            parameter.ref_name_yrs = utils.general.get_name_and_yrs(
+                parameter, ref_data, season
+            )
+            ref = ref_data.get_climo_variable(var, season)
+            ref_site = interpolate_model_output_to_obs_sites(ref, var)
+
+        elif run_type == "model_vs_obs":
+            ref_site = interpolate_model_output_to_obs_sites(None, var)
+        else:
+            raise ValueError("Invalid run_type={}".format(run_type))
+
+        parameter.output_file = (
+            f"{parameter.ref_name}-{parameter.var_id}-{season}-global"
+        )
+        aerosol_aeronet_plot.plot(test, test_site, ref_site, parameter)
+
+    return parameter
+
+
+def interpolate_model_output_to_obs_sites(var, var_id):
+    """Interpolate model outputs (on regular lat lon grids) to observational sites
+
+    :param var: Input model variable, var_id: name of the variable
+    :type var: TransientVariable or NoneType, var_id: str
+    :raises IOError: Invalid variable input
+    :return: interpolated values over all observational sites
+    :rtype: 1-D numpy.array
+
+    """
+    logger.info(
+        "Interpolate model outputs (on regular lat lon grids) to observational sites"
+    )
+    if var_id == "AODABS":
+        aeronet_file = os.path.join(
+            e3sm_diags.INSTALL_PATH, "aerosol_aeronet/aaod550_AERONET_2006-2015.txt"
+        )
+        var_header = "aaod"
+    elif var_id == "AODVIS":
+        aeronet_file = os.path.join(
+            e3sm_diags.INSTALL_PATH, "aerosol_aeronet/aod550_AERONET_2006-2015.txt"
+        )
+        var_header = "aod"
+    else:
+        raise IOError("Invalid variable input.")
+
+    data_obs = pd.read_csv(aeronet_file, dtype=object, sep=",")
+
+    lonloc = np.array(data_obs["lon"].astype(float))
+    latloc = np.array(data_obs["lat"].astype(float))
+    obsloc = np.array(data_obs[var_header].astype(float))
+    # sitename = np.array(data_obs["site"].astype(str))
```

Comment:
```suggestion
```

We can remove commented out code if it is mainly for prototyping/run time testing unless there is a valuable reason to keep it.